### DISCUSSION TITLE
Switching from Composer Installation back to Web-Updates?

### DISCUSSION THREAD

**Discussion by nathanlesage** [wintercms/winter#193] [Q&A] [answered]
Hey y'all,

didn't want to wait any longer and just pushed my OCMS-websites to Winter. It feels great to finally have the Winter logo pop up in the backend. However, I'm curious: I updated by downloading the composer.json and the tests-directory and running composer update. Now, my websites are nothing I do for any clients or something, but purely for myself, and while I'm perfectly comfortable with the command line, I really love that GUI-updater. It's _much_ more comfortable than having to log in to the server every time.

Now, however, I have the composer.json — I don't think that this is that much of a problem, but I wanted to make sure that as soon as the marketplace is operational I can switch back to updating using the GUI – do I simply need to remove the composer.json then, or is there some other magic involved? From what I understand the backend updater simply pulls a pre-built zip file with the vendor-directories already installed, right?

Thanks!

**P.S.:** I read somewhere that I can switch out the old OCMS plugins with their respective Winter-forks by just overwriting all files (but not "removing" them via the backend to retain the data in the database) – but I can't find that anymore. Do I remember this correctly and I can just replace the files?

**Comment by LukeTowers** [comment, marked as answer]
Yes, you can just replace the rainlab plugins with the winter versions in the filesystem and then run winter:up and it will migrate them for you. Note that you'll see an warning message in the plugin management screen, that will be fixed in 1.1.4. 

So long as disableCoreUpdates is not enabled and the update server URL is set correctly, you should be able to update via the backend as soon as that functionality is ready on our marketplace site regardless of whether you have a composer.json or not.

Note that using the GUI will overwrite your composer.json and composer.lock files so don't use the GUI updater if you also use composer for anything else.

**Reply by nathanlesage** [reply]
Thank you again for your help!
